### PR TITLE
[Student][Teacher][MBL-13227] Fix routing for multi-shard courses

### PR DESCRIPTION
**Pull request by CalvinKern**
To test: See ticket

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1006243&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by CalvinKern** [comment]
> Would there be any way to set up our E2E tests such that they encountered the "tilde" IDs?

I'm not sure. Shard instances are pretty strange, I'll see if I can get something up on mobiledev that does this.

**Comment by CalvinKern** [comment]
> Would there be any way to set up our E2E tests such that they encountered the "tilde" IDs?

I'm not seeing any easy way to set this up in our mobile sandbox. I'm guessing it will involve quite a bit of work getting set up (most likely working with IT or toolsmiths). I'll go ahead and create a stub e2e test for this, but don't imagine we could set up anything worthwhile in a reasonable amount of time.

**Comment by joehoag** [comment]
> I'll go ahead and create a stub e2e test for this, but don't imagine we could set up anything worthwhile in a reasonable amount of time.

I wasn't necessarily looking for a dedicated test (though that might help).  I was just thinking that it would be nice if all of our tests that actually hit the server (mainly e2e tests) would encounter these tilde-ids, so that we would have more confidence that we were on top of the issue.

That's what we should create the ticket for -- some way of seeding tilde-ids.

File: libs/canvas-api-2/src/main/java/com/instructure/canvasapi2/utils/APIHelper.kt [hunk 1]
```diff
@@ -205,4 +205,20 @@ object APIHelper {
...
 
+    /**
+     * Parse an ID that references a shard, replaces "~" with the appropriate 0 padding
+     * i.e., converts a sharded ID '12345~4321' to 123450000000004321
+     *
+     * @param id the ID to convert into a long
+     */
+    fun expandTildeId(id: String): String {
+
+        return if (!id.contains("~")) {
```
Comment (Blackchai) [inline libs/canvas-api-2/src/main/java/com/instructure/canvasapi2/utils/APIHelper.kt]: can we reverse the contains check here?

**Pull request merged**